### PR TITLE
[connectors][oauth] fix: Handle revoked/invalid tokens

### PR DESCRIPTION
**Pull request by tdraier**
fixes: https://github.com/dust-tt/tasks/issues/1167

## Description

We have 2 issues when a user revoke its google auth token :  
- first, we get 401 when doing calls with existing token. gdrive connector does not catch the 401 error.
- later, when oauth tries to refresh the token - google drive and confluence may sent a "provider_access_token_refresh_error" with "invalid_grant" when we're trying to refresh an access token which has been revoked by the user.

Fix: 
- In connectors, catching 401 errors sent by google and throw an ExternalOAuthTokenError to pause the connector.
- In core, send a 401 if we get a provider_access_token_refresh_error
- In connectors, handle provider_access_token_refresh_error as token_revoked_error when getting the token and throw ExternalOAuthTokenError

## Risk

false positive, pause connectors when we get an authorization error which was not a real revoke ?

## Deploy Plan

deploy `oauth`
deploy `connectors`

### CODE REVIEW

**Review by Fraggle — APPROVED** [review]
LGTM.

File: connectors/src/lib/oauth.ts [hunk 1]
```diff
@@ -35,7 +35,10 @@ export async function getOAuthConnectionAccessTokenWithThrow({
       "Error retrieving access token"
     );
 
-    if (tokRes.error.code === "token_revoked_error") {
+    if (
+      tokRes.error.code === "token_revoked_error" ||
+      tokRes.error.code === "provider_access_token_refresh_error"
```
[the remark this answers is not in the document]
Comment (spolu) [reply connectors/src/lib/oauth.ts:40]: == ExternalOAuthTokenError should only be emitted on token_revoked_error

Logic should be added to oauth to properly send it when the token is revoked :+1:

File: connectors/src/connectors/google_drive/lib/google_drive_api.ts [hunk 2]
```diff
@@ -29,6 +30,9 @@ export async function getGoogleDriveObject(
 
     return await driveObjectToDustType(file, authCredentials);
   } catch (e) {
+    if ((e as GaxiosError).response?.status === 401) {
```
[the remark this answers is not in the document]
Comment (tdraier) [reply connectors/src/connectors/google_drive/lib/google_drive_api.ts:33]: we could if api calls were centralized in google_drive_api , but we have direct calls to driveClient in different places

File: connectors/src/lib/oauth.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (tdraier) [reply connectors/src/lib/oauth.ts]: looking at the code we should actually get a token_revoked_error (but i don't), reverting that change

**Review by spolu — APPROVED** [review]
LGTM

**Comment by spolu** [comment]
Any insight from the merge?

**Pull request merged**